modules/samples/connectedcup/component/controller/src/main/java/org/coffeeking/controller/service/transport/ConnectedCupMQTTConnector.java

```diff
@@ -0,0 +1,204 @@
+            log.error(errorMsg);
+            throw new TransportHandlerException(errorMsg);
+        }
+
+        String deviceOwner = publishData[0];
+        String deviceId = publishData[1];
+        String resource = publishData[2];
+        String state = publishData[3];
+
+        MqttMessage pushMessage = new MqttMessage();
+        String publishTopic =
+                serverName + File.separator + deviceOwner + File.separator +
+                        ConnectedCupConstants.DEVICE_TYPE + File.separator + deviceId;
+
+        try {
+
+            String actualMessage = resource + ":" + state;
+
+            pushMessage.setPayload(actualMessage.getBytes(StandardCharsets.UTF_8));
+            pushMessage.setQos(DEFAULT_MQTT_QUALITY_OF_SERVICE);
+            pushMessage.setRetained(false);
+
+            publishToQueue(publishTopic, pushMessage);
+
+        } catch (Exception e) {
+            String errorMsg = "Preparing Secure payload failed for device - [" + deviceId + "] of owner - " +
+                    "[" + deviceOwner + "].";
+            log.error(errorMsg);
+            throw new TransportHandlerException(errorMsg, e);
+        }
+    }
+
+
+    @Override
+    public void processIncomingMessage(MqttMessage mqttMessage, String... strings) throws TransportHandlerException {
+        String topic = strings[0];
+
+        String ownerAndId = topic.replace("wso2" + File.separator + "iot" + File.separator, "");
+        ownerAndId = ownerAndId.replace(File.separator + ConnectedCupConstants.DEVICE_TYPE + File.separator, ":");
+        ownerAndId = ownerAndId.replace(File.separator + "connectedcup_publisher", "");
+
+        String owner = ownerAndId.split(":")[0];
+        String deviceId = ownerAndId.split(":")[1];
+
+        String[] messageData = mqttMessage.toString().split(":");
+        Float value = Float.valueOf(messageData[1]);
+        switch (messageData[0]){
+            case "temperature":
+                SensorDataManager.getInstance().setSensorRecord(deviceId, ConnectedCupConstants.SENSOR_TEMPERATURE,
+                                                                String.valueOf(messageData[1]),
+                                                                Calendar.getInstance().getTimeInMillis());
+                ConnectedCupServiceUtils.publishToDAS(owner, deviceId, value);
+                break;
+            case "coffeelevel":
+                SensorDataManager.getInstance().setSensorRecord(deviceId, ConnectedCupConstants.SENSOR_TEMPERATURE,
+                                                                String.valueOf(messageData[1]),
+                                                                Calendar.getInstance().getTimeInMillis());
+                ConnectedCupServiceUtils.publishToDAS(owner, deviceId, value);
+        }
+
+        log.info("Received MQTT message for OWNER: " + owner + " DEVICE.ID: " + deviceId + " | Command: " +
+                 messageData[0] +" " + messageData[1] );
+
```

Comment:
extra new line

Reply:
This file is removed since the device type structure is changed now.